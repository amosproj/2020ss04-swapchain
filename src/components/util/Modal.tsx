import React from "react"
import { ReactComponent as X } from "../../icons/x.svg"
type Props = {
  close: () => void
  open: boolean
  children: any
  title?: string
}

export const Modal = (props: Props) => {
  return (
    <div className={props.open ? "" : "hidden"}>
      <div className="absolute inset-0 bg-gray-500 opacity-75"></div>
      <div className="absolute inset-0 z-10 flex items-center justify-center">
        <div>
          <header className="flex items-center justify-between">
            <h1 className="text-lg font-semibold text-gray-500">{props.title}</h1>
            <X className="w-8 h-8 text-gray-400 duration-200 ease-in-out hover:text-gray-800" onClick={props.close}></X>
          </header>
          <div className="flex flex-col items-center justify-around flex-grow md:flex-row">{props.children}</div>
        </div>
      </div>
    </div>
  )
}
