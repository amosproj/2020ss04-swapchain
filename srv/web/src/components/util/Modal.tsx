import React from "react"
import { ReactComponent as X } from "../../icons/x.svg"
type Props = {
    close: () => void
    open: boolean
    children: any
    title: string
}


export default (props: Props) => {
  
    return (
        <div className={props.open ? "" : "hidden"}>
            <div className="absolute inset-0 bg-gray-500 opacity-75"></div>
            <div className="absolute inset-0 z-10 flex items-center justify-center">
                <div className="p-12 bg-white rounded-sm">
                    <header className="flex items-center justify-between">
                    <h1 className="text-lg font-bold text-gray-500">{props.title}</h1>
                        <X className="w-8 h-8 text-gray-400 duration-200 ease-in-out hover:text-gray-800" onClick={props.close}></X>
                    </header>
                    <div className="flex flex-col items-center justify-around md:flex-row ">
                    {props.children}
                </div>

            </div>
        </div>
        </div>
    )
}

