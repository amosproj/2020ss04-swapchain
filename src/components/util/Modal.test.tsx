import React from "react"
import { Modal } from "./Modal"
import renderer from "react-test-renderer"

it("renders a closed Modal correctly", () => {
  const tree = renderer
    .create(
      <Modal close={() => {}} open={false} title="Hello">
        <h1>Hello</h1>
      </Modal>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("renders an open Modal correctly", () => {
  const tree = renderer
    .create(
      <Modal close={() => {}} open={true} title="Hello">
        <h1>Hello</h1>
      </Modal>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("renders an open Modal without title correctly", () => {
  const tree = renderer
    .create(
      <Modal close={() => {}} open={true}>
        <h1>Hello</h1>
      </Modal>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
