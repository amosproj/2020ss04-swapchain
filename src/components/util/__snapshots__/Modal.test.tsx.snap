// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a closed Modal correctly 1`] = `
<div
  className="hidden"
>
  <div
    className="absolute inset-0 bg-gray-500 opacity-75"
  />
  <div
    className="absolute inset-0 z-10 flex items-center justify-center"
  >
    <div>
      <header
        className="flex items-center justify-between"
      >
        <h1
          className="text-lg font-semibold text-gray-500"
        >
          Hello
        </h1>
        <svg
          className="w-8 h-8 text-gray-400 duration-200 ease-in-out hover:text-gray-800"
          onClick={[Function]}
        >
          x.svg
        </svg>
      </header>
      <div
        className="flex flex-col items-center justify-around flex-grow md:flex-row"
      >
        <h1>
          Hello
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders an open Modal correctly 1`] = `
<div
  className=""
>
  <div
    className="absolute inset-0 bg-gray-500 opacity-75"
  />
  <div
    className="absolute inset-0 z-10 flex items-center justify-center"
  >
    <div>
      <header
        className="flex items-center justify-between"
      >
        <h1
          className="text-lg font-semibold text-gray-500"
        >
          Hello
        </h1>
        <svg
          className="w-8 h-8 text-gray-400 duration-200 ease-in-out hover:text-gray-800"
          onClick={[Function]}
        >
          x.svg
        </svg>
      </header>
      <div
        className="flex flex-col items-center justify-around flex-grow md:flex-row"
      >
        <h1>
          Hello
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders an open Modal without title correctly 1`] = `
<div
  className=""
>
  <div
    className="absolute inset-0 bg-gray-500 opacity-75"
  />
  <div
    className="absolute inset-0 z-10 flex items-center justify-center"
  >
    <div>
      <header
        className="flex items-center justify-between"
      >
        <h1
          className="text-lg font-semibold text-gray-500"
        />
        <svg
          className="w-8 h-8 text-gray-400 duration-200 ease-in-out hover:text-gray-800"
          onClick={[Function]}
        >
          x.svg
        </svg>
      </header>
      <div
        className="flex flex-col items-center justify-around flex-grow md:flex-row"
      >
        <h1>
          Hello
        </h1>
      </div>
    </div>
  </div>
</div>
`;
